worker.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+var commenter = require('./commenter');
+var redisQueue = require('./redisQueue');
+
+var lastCommentTimestamp = 0;
+
+// Process comment tasks
+// Pause for COMMENT_WAIT milleseconds between comment submissions
+redisQueue.process('comment', function(job, done) {
+    var now = Date.now();
+    var waitRemaining = 0;
+
+    var nextCommentTimestamp = lastCommentTimestamp + parseInt(process.env.COMMENT_WAIT);
+
+    if (nextCommentTimestamp > now) {
+        waitRemaining = nextCommentTimestamp - now;
+    }
+
+    setTimeout(function() {
+        commenter.postPullRequestComment(job.data.commentURL, job.data.comment, job.data.filename, job.data.sha, job.data.line);
+        lastCommentTimestamp = Date.now(); // Use a new, fresh value of Date.now()
```

Comment:
Yeah, great point Blixa! I like that you accounted for this in your pull request. We should definitely do more to handle rejected comments.

Redis has a handy feature for re-attempting failed jobs with exponentially longer delays. I updated the code to use that. It doesn't read the `Retry-After` header, so it may end up commenting a bit later than it really needs to, but one benefit with the exponential retry approach is that it will be able to handle GitHub downtime and other cases where the `Retry-After` header isn't available.